src/views.rs

```diff
@@ -842,3 +842,101 @@ async fn show_high_scores(
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::connection::Receiver;
+
+    #[tokio::test]
+    async fn test_name_entering_on_windows_cmd_exe() {
+        let mut client = Client::new(123, Receiver::Test("WindowsUsesCRLF\r\n".to_string()));
+        let result = ask_name(&mut client, Arc::new(Mutex::new(HashSet::new()))).await;
+        assert!(result.is_ok());
+        assert_eq!(client.get_name(), Some("WindowsUsesCRLF"));
+    }
+
+    #[tokio::test]
+    async fn test_forgot_stty_raw() {
+        let mut client = Client::new(123, Receiver::Test("Oops\n".to_string()));
+        let result = ask_name(&mut client, Arc::new(Mutex::new(HashSet::new()))).await;
+        assert!(result.is_err());
+        assert_eq!(client.get_name(), None);
+
+        let text = client.render_data.lock().unwrap().buffer.get_text();
+        assert!(text.contains("Your terminal doesn't seem to be in raw mode"));
+    }
+
+    #[tokio::test]
+    async fn test_entering_name_on_raw_linux_terminal() {
+        let mut client = Client::new(123, Receiver::Test("linux_usr\r".to_string()));
+        let result = ask_name(&mut client, Arc::new(Mutex::new(HashSet::new()))).await;
+        assert!(result.is_ok());
+        assert_eq!(client.get_name(), Some("linux_usr"));
+    }
+
+    #[tokio::test]
+    async fn test_long_name() {
+        let mut client = Client::new(
+            123,
+            Receiver::Test("VeryVeryLongNameGoesHere\r".to_string()),
+        );
+        let result = ask_name(&mut client, Arc::new(Mutex::new(HashSet::new()))).await;
+        assert!(result.is_ok());
+        assert_eq!(client.get_name(), Some("VeryVeryLongNam"));
+
+        // Name should show up as truncated to the user entering it
+        let text = client.render_data.lock().unwrap().buffer.get_text();
```

Comment:
Maybe `get_text()` could be a very convenient convenience method, named `text()` and directly on the client. Mantaray has something very similar:

```python3
    # for tests
    def text(self) -> str:
        return self.get_current_view().textwidget.get("1.0", "end - 1 char")
```